docs/src/basics/solution.md

```diff
@@ -181,23 +181,38 @@ SciMLBase.DEStats
 
 ## [Return Codes (RetCodes)](@id retcodes)
 
-The solution types have a `retcode` field which returns a symbol signifying the
-error state of the solution. The retcodes are as follows:
+The solution types have a `retcode` field which returns an enum value signifying the
+error state of the solution. Return codes are now implemented as an enum using EnumX.jl
+rather than symbols.
 
-  - `:Default`: The solver did not set retcodes.
-  - `:Success`: The integration completed without erroring or the steady state solver
+To check if a solution was successful, use:
+```julia
+SciMLBase.successful_retcode(sol)
+```
+
+!!! warning
+    Previous iterations of the interface suggested using `sol.retcode == :Success`, 
+    however, that is now not advised. Use `SciMLBase.successful_retcode(sol)` instead.
```

Comment:
```suggestion
    however, that is now not advised because there are more than one return code that can be interpreted
    as successful. For example, `Terminated` is a successful run to a manual termination, and would be missed
    if only checking for Success. Therefore we highly recommend you use `SciMLBase.successful_retcode(sol)` instead.
```